app/controllers/my_facilities/drug_stocks_controller.rb

```diff
@@ -92,10 +92,10 @@ def set_region
   end
 
   def blocks_to_display
-    if @selected_facility_sizes != @facility_sizes || @selected_zones != @zones
-      Region.none
-    else
+    if @selected_facility_sizes == @facility_sizes && @selected_zones == @zones
       @query.blocks.order(:name)
+    else
+      Region.none
```

Comment:
Much better😅

Reply:
@harimohanraj89 spotted it in our original PR 🙏